kyaml/pathutil/pathutil.go

```diff
@@ -9,20 +9,27 @@ import (
 	"strings"
 )
 
-// SubDirsWithFile takes the root directory path and returns all the paths of
-// sub-directories which contain file with input fileName including itself
-func SubDirsWithFile(root, fileName string) ([]string, error) {
+// DirsWithFile takes the root directory path and returns all the paths of
+// sub-directories(including itself) which contain file with input fileName
+// at top level if recurse is true
+func DirsWithFile(root, fileName string, recurse bool) ([]string, error) {
 	var res []string
+	if !recurse {
+		// check if the file with fileName is present in root and return it
+		// else return empty list
+		_, err := os.Stat(filepath.Join(root, fileName))
+		if !os.IsNotExist(err) {
+			res = append(res, root)
+		}
+		return res, nil
+	}
 	err := filepath.Walk(root,
-		func(path string, info os.FileInfo, err error) error {
+		func(filePath string, info os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
-			if strings.HasSuffix(path, fileName) {
-				if root == "." {
-					path = root + "/" + path
-				}
-				res = append(res, path)
+			if strings.HasSuffix(filePath, fileName) {
```

Comment:
I think we should use a function from the `filepath` package here to get the name of the file and then check if the filename match what we are looking for.

Reply:
Done.